### PR TITLE
adding participation data type

### PR DESCRIPTION
**Pull request by c-monkey**
Pull requests into Health Data Standards require the following. Submitter and reviewer should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable ("N/A") and :white_check_mark:.
 
**Submitter:**
- [X] This pull request describes why these changes were made.
- [X] This PR is into the correct branch.
- [X] Internal ticket for this PR: https://jira.mitre.org/browse/BONNIE-1377
- [x] Internal ticket links back to this PR
- [X] Code diff has been done and been reviewed (it **does not** contain: additional white space, not applicable code changes, debug statements, etc.)
- [X] Tests are included and test edge cases
- [X] Tests have been run locally and pass
- [X] Code coverage has not gone down and all code touched or added is covered.

Branch | Coverage
-- | --
bonnie-v2.1 | 1709 / 6975 / 7699 LOC (90.6%) covered. <br> 2692 / 7797 LOC (34.53%) covered.
bonnie_1025_participation_data_type | 6978 / 7702 LOC (90.6%) covered. <br> 2695 / 7800 LOC (34.55%) covered.

 
**Cypress Reviewer:**
 
Name: @dczulada 
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [ ] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code
 
**Bonnie Reviewers:**
 
Name: @losborne 
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

Name: @holmesie
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

### CODE REVIEW

**Review by dczulada — APPROVED** [review]

File: lib/hqmf-model/data_criteria.json [hunk 1]
```diff
@@ -1042,5 +1042,14 @@
       "sub_category":"",
       "hard_status":false,
       "patient_api_function":"allergies",
-      "not_supported":false}
+      "not_supported":false},
+    "participation":{
+        "title":"participation",
+        "category":"participation",
```
Comment (lukoktonos) [inline lib/hqmf-model/data_criteria.json]: This field should be plural, along with "patient_api_function" below and it's corresponding entry in cqlpatient.js cql_qdm_patientapi and health-data-standards:lib/health-data-standards/models/record.rb

We discussed this a bit at standup without necessarily a clear decision, but after re-considering, I now strongly support 'participations' as the plural form, for clarity and consistency.

Comment (holmesie) [reply lib/hqmf-model/data_criteria.json]: I agree with adding an `s` for both of these, as that is the behavior of the Rails Inflector (http://api.rubyonrails.org/classes/ActiveSupport/Inflector.html#method-i-pluralize):
```
irb(main):003:0> "participation".pluralize
=> "participations"
```
Being able to go from the singular to plural using `pluralize` is nifty, and actually expected in my converters, so I'd like to keep that.

**Review by holmesie — CHANGES_REQUESTED** [review]
Are we happy with just calling this `Participant`? I'm not sure where that is coming from in the QDM spec. Is the actual QDM data type `Patient Characteristic Clinical Trial Participant`? I suppose it's not a huge deal HDS wise, as HDS is already so divergent from QDM in the first place, and that my new QDM based models have a `PatientCharacteristicClinicalTrialParticipant` model (so this won't be an issue in the near future) - but I'd just like to make sure we are fine with that.

**Review by lukoktonos — APPROVED** [review]

**Review by holmesie — APPROVED** [review]
